### DISCUSSION TITLE
"Expert partitioner" and "Swap"

### DISCUSSION THREAD

**Discussion by RafaelLinux** [agama-project/agama#845] [Ideas]
On my machine, I have installed both openSUSE Tumbleweed and RegataOS (based on openSUSE) and Fedora. They all share the same SWAP, whose size is proportional to the RAM of my computer, so the location and size of my SWAP partition is just the right one for the characteristics of my hardware. I give this example, because I don't understand why not to reuse a SWAP partition that already exists, when it is more than likely that this partition is the appropriate one for this scenario and does not require anything else from the user.

On the other hand, in Agama I can't find something very important for users (advanced or not), which is the "Expert Partitioning", where you can choose "Start with Existing Partitions" and later, you can import the users of the previous partitions.

**Comment by ancorgs** [comment]
Regarding swap reuse. We disabled it for the time being because the current Agama UI, which presents the list of file-system that will be created, doesn't have a way to represent that a given file-system will be located in a reused device. That makes the whole thing pretty confusing.

https://yast.opensuse.org/assets/images/blog/2023-08-21/storage.png

We would like to add the ability in the future to specify that a given file-system must be created in an already existing partition/LV. Then we may bring back swap reusing since it will be more clear what is happening and the user may even decide to disable that reusing.

Regarding the Expert Partitioner, we don't plan to bring it back in the exact form it exists in YaST.  You can know more about how we envision the whole storage configuration at https://github.com/openSUSE/agama/blob/master/doc/storage_ui.md

**Comment by RafaelLinux** [comment]
Sincerely, I believe that advanced functions such as the mentioned importing of previous partitions (I guess reading the "fstab" file) as well as users and their passwords should not be lost. If we lose those features, we start having an installer that other distros will have nothing to envy (let's say, "Calamares").

I think that, although it is more complex, advanced installation features should not be lost, but allow - as it did so far the Yast installer - that the user is what he wants to do, as simple as allowing two options from the beginning of the installation, one would be "Simple installation" and another "Advanced installation". It will be the user who will have chosen the complexity from that point on. 

Some people refer to videos where they analyze openSUSE and these "analysts" mention the "complexity" of the installer. Usually these "analyst" opinions are made from the perspective of "distrohoppers" who are not regular users of a particular distribution, but are just "experimenting". When performing an installation, this kind of users do not care if the data partition will be in another partition of the same system disk, or the file system to be used in the data partition, or if it will include an additional parameter to perform on-the-fly compression, or if we want to have a single EFI partition to have several distros. 

For regular users of openSUSE who use it both at work and at home, we do care about being able to reinstall openSUSE so that we can quickly recover mounted local and remote drives (from a NAS, for example) as well as the optional parameters we had already assigned. In my case, I always enable ZSTD compression on all SSD disks, as the installer never asks about it (not even in the advanced options). Because, after all, how advanced is the openSUSE installer if it keeps the current features? I would understand it to be "advanced" or "expert" if it showed the user many more options, such as the one mentioned about compression, or if the partition is read-only, the type of encryption to be used, if it allows execution, etc.

For all these reasons, please, let's not lose the features that make openSUSE installer the most flexible and convenient installer available.

**Reply by ancorgs** [reply]
Many of the features you mention in your comment will be kept. Just implemented in a different way. But it's a long process and we are prioritizing those features that are useful for a broader set of people.

**Reply by RafaelLinux** [reply]
Thank you very much for the information. It's a relief to know that we won't lose these advanced features and that, for me and many other less loyal openSUSE users, they are essential to maintain the upgrade cycles of LEAP machines or even other Linux distributions that want to switch to openSUSE, keeping the system partition structure they already had and the information of the data partitions intact, to give a couple of examples.

**Comment by imobachgs** [comment]
Hi @RafaelLinux,

Thanks for your comments! Given that @ancorgs already explained that the plan is to keep most of those features (but implementing them differently), we can close this discussion by now. Of course, please don't hesitate to reopen it if you need to.

**Reply by RafaelLinux** [reply]
Perfect. I should have closed it after reading your last message  :(

Thank you for all your effort to give us a better installer for all.

**Reply by imobachgs** [reply]
No problem! Thanks for bringing up this topic. 😊
